### PR TITLE
Add spoilered rolls; s#dX syntax

### PR DESCRIPTION
**Pull request by Nichlas-Schmidt**

### CODE REVIEW

File: packages/bot/src/common/behaviors/dice/dice.behavior.base.ts
Comment (Nichlas-Schmidt) [inline packages/bot/src/common/behaviors/dice/dice.behavior.base.ts:184]: this line can be removed

File: packages/bot/src/common/behaviors/dice/dice.behavior.base.ts
Comment (david-sharer) [inline packages/bot/src/common/behaviors/dice/dice.behavior.base.ts:159]: It looks like this might be able to handle both spoilered and unspoilered results with some minor tweaks.
I think that's beyond the scope of this PR tho.

File: packages/bot/src/common/behaviors/dice/dice.behavior.base.ts
Comment (david-sharer) [inline packages/bot/src/common/behaviors/dice/dice.behavior.base.ts:180]: Note for me later: See if this has interactions with formatting

File: packages/bot/src/common/behaviors/dice/dice.behavior.base.ts
Comment (david-sharer) [inline packages/bot/src/common/behaviors/dice/dice.behavior.base.ts:182]: Note for me later: indentation here seems weird. Not sure if that's related to the PR viewer, but either way I should probably set up a default code formatter.

**Review by david-sharer — COMMENTED** [review]
Note for me later: Add docs + version + changelog after merging (or maybe there's a way for me to suggest revisions to this? I'm not super familar with the Github PR process)

File: packages/bot/src/common/behaviors/dice/dice.behavior.base.ts
Comment (david-sharer) [reply packages/bot/src/common/behaviors/dice/dice.behavior.base.ts:180]: It did. Resolving this well would require passing another arg to the `buildMessage` function to determine if we escape `|` characters.

image

File: packages/bot/src/common/behaviors/dice/dice.behavior.base.ts
Comment (david-sharer) [reply packages/bot/src/common/behaviors/dice/dice.behavior.base.ts:182]: https://github.com/rollem-discord/rollem-discord/issues/228

File: packages/bot/src/common/behaviors/dice/dice.behavior.base.ts
Comment (david-sharer) [reply packages/bot/src/common/behaviors/dice/dice.behavior.base.ts:180]: https://github.com/rollem-discord/rollem-discord/issues/229

**Review by david-sharer — APPROVED** [review]
Looks fine to merge. One thing that could be cleaned up, but doesn't hurt anything.

I made a couple tickets for potential patches later.
One to avoid redundancy in `s#` and `\d#` handlers (and support multi+spoiler rolls).
Another to avoid spoiler-tag conflicts in the label.
IMO, neither is important enough to block the PR.

**Comment by david-sharer** [comment]
Resolves #225 #200

**Pull request merged**
